Use new image API to save/show image of an entry

AdityaKhatri opened the pull request:
Depends on https://github.com/the-deep/server/pull/669

Depends on https://github.com/the-deep/server/pull/652

AdityaKhatri committed:
Use new image API to save/show image of an entry

tnagorra commented on src/components/viewer/ImagesGrid/index.js:
```diff
@@ -9,6 +10,36 @@ import ModalBody from '#rscv/Modal/Body';
+            onDragStart={handleDragStart}
+        />
+    );
+}
```
Image should either have a propTypes definition or should be written in TS

tnagorra commented on src/entities/editEntries.js:
```diff
@@ -165,9 +166,15 @@ export const createEntry = ({
     const keysToPick = [
         'excerpt',
         'image',
+        'imageRaw',
+        'leadImage',
```
This seems weird.
1. What is leadImage on entries? Where is this used?
2. What is the difference between image and imageRaw? Do we need both?

AdityaKhatri replied:
1. LeadImage is the `id` of image coming from lead if we are dragging from the `images column.
2. ImageRaw is the blob we save before creating of image file
3. Image is the generated file URL

tnagorra commented on src/schema/entries.js:
```diff
@@ -33,7 +33,7 @@ const entrySchema = [];
             analysisFramework: { type: 'uint', required: true },
 
-            image: { type: 'string' },
+            image: { type: 'number' },
```
Shouldn't we add imageRaw imageDetails here?

tnagorra replied:
...

tnagorra commented on src/views/EditEntries/Overview/LeftPane/EntriesList/EntryItem/index.js:
```diff
@@ -92,6 +94,8 @@ const EntryItem = (props) => {
                 <EntryPreview
                     entryType={entryType}
                     image={image}
+                    imageRaw={imageRaw}
```
Do we even need to pass `image` here?

AdityaKhatri replied:
No, I'll remove it

tnagorra replied:
When?

tnagorra commented on src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:
```diff
@@ -74,6 +77,8 @@ export default class EntriesList extends React.PureComponent {
             entryServerId: currentEntryServerId,
             entryType,
             image,
+            imageRaw,
```
Check if image should be passed

tnagorra replied:
...

AdityaKhatri replied:
It is not requred

tnagorra commented on src/views/EditEntries/Overview/LeftPane/EntriesList/index.js:
```diff
@@ -54,8 +54,11 @@ export default class EntriesList extends React.PureComponent {
+
         const {
             entryType,
+            imageRaw,
```
Check if image should be passed

tnagorra replied:
...

AdityaKhatri replied:
not required

tnagorra commented on src/views/EditEntries/Overview/index.js:
```diff
@@ -194,7 +194,13 @@ export default class Overview extends React.PureComponent {
+            dropped,
+            imageDetails,
+        } = excerptData;
+        console.warn('i am here', imageDetails);
```
No console.warn for debugging.

tnagorra commented on src/views/EditEntries/WidgetFaram/index.js:
```diff
@@ -446,10 +452,12 @@ export default class WidgetFaram extends React.PureComponent {
             excerpt,
             droppedExcerpt,
             image,
```
Check if we need to pass image here

tnagorra commented on src/views/EditEntries/WidgetFaram/index.js:
```diff
@@ -466,6 +474,8 @@ export default class WidgetFaram extends React.PureComponent {
                 excerpt,
                 droppedExcerpt,
                 image,
```
Check if we need to pass image here

tnagorra commented on src/widgets/tagging/Excerpt/index.js:
```diff
@@ -186,12 +189,16 @@ export default class Excerpt extends React.PureComponent {
     renderExcerptImage = () => {
-        const { image } = this.props;
+        const {
+            image,
```
Do we even need this?

tnagorra committed:
Use url instead of id in InternalGallery

- Use displayPictureUrl field instead of displayPicture field
- Use attachment.file instead of attachment.id
- Remove file fetch logic from InternalGallery
- Remove onMimeTypeGet function from InternalGallery
- Remove pending logic from InternalGallery

tnagorra commented on src/components/general/Navbar/NavDrop/index.js:
```diff
@@ -103,8 +103,8 @@ DropItem.defaultProps = {
     activeCountryId: activeCountryIdFromStateSelector(state),
-    activeUser: activeUserSelector(state),
-    userInformation: currentUserInformationSelector(state),
+    activeUser: activeUserSelector(state), // from auth
+    userInformation: currentUserInformationSelector(state), // from users
```
No need to keep this comment

tnagorra replied:
...

tnagorra commented on src/components/viewer/Attachment/index.js:
```diff
@@ -91,7 +94,7 @@ export default class Attachment extends React.PureComponent {
                     return (
                         <InternalGallery
-                            galleryId={attachment && attachment.id}
+                            attachmetn={attachment}
```
```suggestion
                            attachment={attachment}
```

AdityaKhatri committed:
Fix review comments

samshara committed:
Use Avatar component instead of DisplayPicture

tnagorra committed:
Maintain 'files' list for uploaded files in assessment

- Add ExternalGalleryModal

AdityaKhatri committed:
Use Avatar in RecentActivity

AdityaKhatri committed:
Fix images grid image preview

AdityaKhatri committed:
Refresh flie url on lead upload

AdityaKhatri committed:
Restrict lead attachement fetch on lead edit modal

AdityaKhatri committed:
Add missing schema items

AdityaKhatri committed:
Fix image in ary entries list
